### ISSUE TITLE
FPP 2.5 DNS Stops Working

### ISSUE THREAD

**Issue by dennismccreery** [FalconChristmas/fpp#463]
> [Quote from: gocats7 on November 22, 2018, 09:12:43 PM](https://falconchristmas.com/forum/index.php/topic,9961.msg90711.html#msg90711)
> Seems to be some issue with DNS.  After upgrading from 2.4.2, fpp couldnt resolve 
> github when trying to do a manual update so I went to terminal and there was no dns 
> resolution, but I could hit things via IP  which confirms internet connectivity.  
> Rebooted, no change.  I have static network settings so I went and clicked update dns 
> then restart dns (made no changes) and from terminal DNS worked.  
> Rebooted, it didn't work again so I repeated steps and it then worked again.

I had the same issue with the BBB version of 2.5.  Noticed that my show wasn't starting tonight (ran fine for the past two nights) and I went to the Status page, it was showing the wrong date/time (20 days before now).  Checked DNS settings, they looked okay (8.8.8.8 and 8.8.4.4).  Could ping the primary DNS ( 8.8.8.8 ) just fine.  Went to the Help/About page to see if there were any patch updates, saw this message:

```
FPP Manual Update

Stopping fppd...
Pulling in updates... 
Can not access github, unable to pull git updates
Restarting fppd...
Update Complete.
```

Tried restarting DNS, then restarting network, no joy.  Then rebooted and it's still the same, can't seem to access the network even though ping works fine.  Here's the syslog messages around the time that the date/time changed:

```
Nov 20 09:25:01 FPP CRON[6613]: (root) CMD (command -v debian-sa1 > /dev/null && debian-sa1 1 1)
Nov 20 09:35:01 FPP CRON[14187]: (root) CMD (command -v debian-sa1 > /dev/null && debian-sa1 1 1)
Nov 20 09:39:01 FPP CRON[17230]: (root) CMD (  [ -x /usr/lib/php/sessionclean ] && if [ ! -d /run/systemd/system ]; then /usr/lib/php/sessionclean; fi)
Nov 20 09:39:12 FPP systemd[1]: Starting Clean php session files...
Nov 20 09:39:14 FPP systemd[1]: Started Clean php session files.
Nov 20 09:45:01 FPP CRON[21818]: (root) CMD (command -v debian-sa1 > /dev/null && debian-sa1 1 1)
Nov  3 10:17:15 FPP kernel: [   10.722407] musb-hdrc musb-hdrc.1: Babble
Nov  3 10:17:15 FPP kernel: [   10.726466] musb-hdrc musb-hdrc.1: Babble
Nov  3 10:17:15 FPP kernel: [   10.730525] musb-hdrc musb-hdrc.1: Babble

```
Not sure where to go from here.  Tried powering the system down and then up again, still the same.  Seems it can no longer get to the internet DNS.  Here's what I see from the shell:

```
fpp@FPP:~$ ping 192.168.29.1                                                                                                                                                     
PING 192.168.29.1 (192.168.29.1) 56(84) bytes of data.                                                                                                                           
64 bytes from 192.168.29.1: icmp_seq=1 ttl=64 time=0.823 ms                                                                                                                      
64 bytes from 192.168.29.1: icmp_seq=2 ttl=64 time=0.931 ms                                                                                                                      
64 bytes from 192.168.29.1: icmp_seq=3 ttl=64 time=0.963 ms                                                                                                                      
64 bytes from 192.168.29.1: icmp_seq=4 ttl=64 time=0.903 ms                                                                                                                      
^C                                                                                                                                                                               
--- 192.168.29.1 ping statistics ---                                                                                                                                             
4 packets transmitted, 4 received, 0% packet loss, time 3004ms                                                                                                                   
rtt min/avg/max/mdev = 0.823/0.905/0.963/0.051 ms                                                                                                                                
fpp@FPP:~$ ping 8.8.8.8
PING 8.8.8.8 ( 8.8.8.8 ) 56(84) bytes of data.                                                                                                                                     
64 bytes from 8.8.8.8: icmp_seq=1 ttl=44 time=24.2 ms                                                                                                                            
64 bytes from 8.8.8.8: icmp_seq=2 ttl=44 time=24.4 ms                                                                                                                            
64 bytes from 8.8.8.8: icmp_seq=3 ttl=44 time=17.6 ms                                                                                                                            
64 bytes from 8.8.8.8: icmp_seq=4 ttl=44 time=22.7 ms                                                                                                                            
^C                                                                                                                                                                               
--- 8.8.8.8 ping statistics ---                                                                                                                                                  
4 packets transmitted, 4 received, 0% packet loss, time 3004ms                                                                                                                   
rtt min/avg/max/mdev = 17.639/22.253/24.418/2.751 ms                                                                                                                             
fpp@FPP:~$ 
fpp@FPP:~$ ping www.google.com
ping: www.google.com: Temporary failure in name resolution   
```

Looks like my system's resolv.conf is getting wiped somehow:

fpp@FPP~$ cat /etc/resolv.conf
cat: /etc/resolv.conf: No such file or directory                                                                                                                                 
fpp@FPP:~$ 

Appears to be similar to the problem discussed in this thread: https://falconchristmas.com/forum/index.php/topic,7847.15.html

Logs are attached:
[FPP_Logs_20161103.1720.zip](https://github.com/FalconChristmas/fpp/files/2612839/FPP_Logs_20161103.1720.zip)

**Comment by dennismccreery** [comment]
Looks like Master also has problems with the Scheduler -- when I start FPP up in the middle of a schedule, it doesn't start the schedule by itself.  Downgraded to 2.4 and it seems to work.

**Comment by andyaz** [comment]
DNS issue also present on rPi.

**Comment by sithdeath** [comment]
This might already be known, but while looking at the routing table on the rpi3, it is adding the ip of the default route to the eth0 interface, which after removing, and setting the default route, it seems to work..

Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
default         192.168.70.1    0.0.0.0         UG    0      0        0 eth0
192.168.0.0     0.0.0.0         255.255.255.0   U     0      0        0 tether
192.168.1.0     0.0.0.0         255.255.255.0   U     0      0        0 eth0
192.168.70.0    0.0.0.0         255.255.255.0   U     0      0        0 wlan0
192.168.70.1    0.0.0.0         255.255.255.255 UH    0      0        0 wlan0
**192.168.70.1    0.0.0.0         255.255.255.255 UH    0      0        0 eth0**

root@myhost:/home/fpp# route del -host 192.168.0.1 dev eth0

Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
default         192.168.70.1    0.0.0.0         UG    0      0        0 eth0
192.168.0.0     0.0.0.0         255.255.255.0   U     0      0        0 tether
192.168.1.0     0.0.0.0         255.255.255.0   U     0      0        0 eth0
192.168.70.0    0.0.0.0         255.255.255.0   U     0      0        0 wlan0
192.168.70.1    0.0.0.0         255.255.255.255 UH    0      0        0 wlan0

root@myhost:/home/fpp# ping www.yahoo.com
ping: www.yahoo.com: Temporary failure in name resolution

After a few hours, the default route disappears by itself:

Destination     Gateway         Genmask         Flags Metric Ref    Use Iface
192.168.0.0     0.0.0.0         255.255.255.0   U     0      0        0 tether
192.168.1.0     0.0.0.0         255.255.255.0   U     0      0        0 eth0
192.168.70.0    0.0.0.0         255.255.255.0   U     0      0        0 wlan0
192.168.70.1    0.0.0.0         255.255.255.255 UH    0      0        0 wlan0

And then when I add it back in:
root@myhost:/home/fpp# route add default gw 192.168.70.1

Ping is working again and so is the update from the about page :-)

Sorry for the wrong formatting as this is my first post on Github...

**Comment by dkulp** [comment]
Fixed with 2.5.1
